In `magicleap` build:

```rust
error[E0046]: not all trait items implemented, missing: `on_alert`
   --> ports/libmlservo/src/lib.rs:338:1
    |
338 | impl HostTrait for HostCallbacks {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `on_alert` in implementation
    |
    = note: `on_alert` from trait: `fn(&Self, std::string::String)`

error: aborting due to previous error
```

--- a/ports/libmlservo/src/lib.rs
+++ b/ports/libmlservo/src/lib.rs
@@ -347,6 +347,7 @@ impl HostTrait for HostCallbacks {
         MakeCurrent(self.disp, self.surf, self.surf, self.ctxt);
     }
 
+    fn on_alert(&self, _message: String) {}
     fn on_load_started(&self) {}
     fn on_load_ended(&self) {}
     fn on_title_changed(&self, _title: String) {}

--- a/ports/libsimpleservo/api/src/lib.rs
+++ b/ports/libsimpleservo/api/src/lib.rs
@@ -89,6 +89,8 @@ pub trait HostTrait {
     /// Will be called before drawing.
     /// Time to make the targetted GL context current.
     fn make_current(&self);
+    /// javascript window.alert()
+    fn on_alert(&self, msg: String);
     /// Page starts loading.
     /// "Reload button" should be disabled.
     /// "Stop button" should be enabled.
@@ -521,6 +523,7 @@ impl ServoGlue {
                 },
                 EmbedderMsg::Alert(message, sender) => {
                     info!("Alert: {}", message);
+                    self.callbacks.host_callbacks.on_alert(message);
                     let _ = sender.send(());
                 },
                 EmbedderMsg::AllowOpeningBrowser(response_chan) => {

--- a/ports/libsimpleservo/capi/src/lib.rs
+++ b/ports/libsimpleservo/capi/src/lib.rs
@@ -38,6 +38,7 @@ where
 pub struct CHostCallbacks {
     pub flush: extern "C" fn(),
     pub make_current: extern "C" fn(),
+    pub on_alert: extern "C" fn(message: *const c_char),
     pub on_load_started: extern "C" fn(),
     pub on_load_ended: extern "C" fn(),
     pub on_title_changed: extern "C" fn(title: *const c_char),
@@ -344,6 +345,14 @@ impl HostTrait for HostCallbacks {
     fn make_current(&self) {
         debug!("make_current");
         (self.0.make_current)();
+    }
+
+    fn on_alert(&self, message: String) {
+        debug!("on_alert");
+        let message = CString::new(message).expect("Can't create string");
+        let msg_ptr = message.as_ptr();
+        mem::forget(message);
+        (self.0.on_alert)(msg_ptr);
     }
 
     fn on_load_started(&self) {

--- a/ports/libsimpleservo/jniapi/src/lib.rs
+++ b/ports/libsimpleservo/jniapi/src/lib.rs
@@ -375,6 +375,23 @@ impl HostTrait for HostCallbacks {
             .unwrap();
     }
 
+    fn on_alert(&self, message: String) {
+        debug!("on_alert");
+        let env = self.jvm.get_env().unwrap();
+        let s = match new_string(&env, &message) {
+            Ok(s) => s,
+            Err(_) => return,
+        };
+        let s = JValue::from(JObject::from(s));
+        env.call_method(
+            self.callbacks.as_obj(),
+            "onAlert",
+            "(Ljava/lang/String;)V",
+            &[s],
+        )
+        .unwrap();
+    }
+
     fn on_load_started(&self) {
         debug!("on_load_started");
         let env = self.jvm.get_env().unwrap();

--- a/support/android/apk/servoapp/src/main/java/org/mozilla/servo/MainActivity.java
+++ b/support/android/apk/servoapp/src/main/java/org/mozilla/servo/MainActivity.java
@@ -130,6 +130,10 @@ public class MainActivity extends Activity implements Servo.Client {
     }
 
     @Override
+    public void onAlert(String message) {
+    }
+
+    @Override
     public void onLoadStarted() {
         mReloadButton.setEnabled(false);
         mStopButton.setEnabled(true);

--- a/support/android/apk/servoview/src/main/java/org/mozilla/servoview/JNIServo.java
+++ b/support/android/apk/servoview/src/main/java/org/mozilla/servoview/JNIServo.java
@@ -93,6 +93,8 @@ public class JNIServo {
 
         void makeCurrent();
 
+        void onAlert(String message);
+
         void onAnimatingChanged(boolean animating);
 
         void onLoadStarted();

--- a/support/android/apk/servoview/src/main/java/org/mozilla/servoview/Servo.java
+++ b/support/android/apk/servoview/src/main/java/org/mozilla/servoview/Servo.java
@@ -169,6 +169,8 @@ public class Servo {
     }
 
     public interface Client {
+        void onAlert(String message);
+
         boolean onAllowNavigation(String url);
 
         void onLoadStarted();
@@ -226,6 +228,10 @@ public class Servo {
         public void makeCurrent() {
             // Up to the callback to execute this in the right thread
             mGfxCb.makeCurrent();
+        }
+
+        public void onAlert(String message) {
+            mRunCallback.inUIThread(() -> mClient.onAlert(message));
         }
 
         public void onShutdownComplete() {

